File: emedia/src/main/java/com/xjl/emedia/bean/MediaPickerBean.java
package com.xjl.emedia.bean;

import androidx.annotation.NonNull;

import java.io.Serializable;

/**
 * Created by x33664 on 2018/11/8.
 */

public class MediaPickerBean implements Serializable, Comparable<MediaPickerBean> {

    public long size;
    public String mediaFilePath;
    public boolean isPicked = false;
    public int type;
    public int data;
    public String duration;

    public MediaPickerBean() {
    }

    ;

    public MediaPickerBean(int createTime, String mediaFilePath, int type, int data) {
        this.size = createTime;
        this.mediaFilePath = mediaFilePath;
        this.data = data;
        this.type = type;
    }

    public long getSize() {
        return size;
    }

    public void setSize(long size) {
        this.size = size;
    }

    public String getMediaFilePath() {
        return mediaFilePath;
    }

    public void setMediaFilePath(String mediaFilePath) {
        this.mediaFilePath = mediaFilePath;
    }

    public boolean isPicked() {
        return isPicked;
    }

    public void setPicked(boolean picked) {
        isPicked = picked;
    }

    public int getData() {
        return data;
    }

    public void setData(int data) {
        this.data = data;
    }

    public String getDuration() {
        return duration;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }


    @Override
    public int compareTo(@NonNull MediaPickerBean o) {
        return o.data - this.data;
    }
}

File: emedia/src/main/java/com/xjl/emedia/bean/MediaPickerBean.kt
package com.xjl.emedia.bean

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by x33664 on 2018/11/8.
 */
class MediaPickerBean : Parcelable, Comparable<MediaPickerBean> {
    
    var size: Long = 0
    
    var mediaFilePath: String = ""
    
    var isPicked = false
    
    var type = 0
    
    var data = 0
    
    var duration: String = ""

    constructor()
    constructor(createTime: Int, mediaFilePath: String, type: Int, data: Int) {
        size = createTime.toLong()
        this.mediaFilePath = mediaFilePath
        this.data = data
        this.type = type
    }

    protected constructor(parcel: Parcel) {
        size = parcel.readLong()
        mediaFilePath = parcel.readString()?:""
        isPicked = parcel.readByte().toInt() != 0
        type = parcel.readInt()
        data = parcel.readInt()
        duration = parcel.readString()?:""
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(size)
        dest.writeString(mediaFilePath)
        dest.writeByte((if (isPicked) 1 else 0).toByte())
        dest.writeInt(type)
        dest.writeInt(data)
        dest.writeString(duration)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun compareTo(o: MediaPickerBean): Int {
        return o.data - data
    }

    companion object CREATOR : Parcelable.Creator<MediaPickerBean> {
        override fun createFromParcel(parcel: Parcel): MediaPickerBean {
            return MediaPickerBean(parcel)
        }

        override fun newArray(size: Int): Array<MediaPickerBean?> {
            return arrayOfNulls(size)
        }
    }


}
